// Copyright 2010-2022 Google LLC
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_package = "com.google.ortools.util";
option java_multiple_files = true;
option csharp_namespace = "Google.OrTools.Util";

package operations_research;

// A "three-way" boolean: unspecified, false or true.
//
// We don't use the value of 1 to increase the chance to catch bugs: eg. in
// python, a user may set a proto field of this type enum to a boolean value
// without type checks, if they set it to True, the proto validity code will
// catch it (because it'll be cast to 1, which is an invalid enum value).
// Note that if the user sets if to False (i.e. 0), it will be caught by the
// routing library's parameter validity check too.
enum OptionalBoolean {
  BOOL_UNSPECIFIED = 0;
  BOOL_FALSE = 2;
  BOOL_TRUE = 3;
}
